safety/scan/main.py

```diff
@@ -196,8 +203,43 @@ def save_report_as(scan_type: ScanType, export_type: ScanExport, at: Path, repor
     with open(at, 'w+') as report_file:
         report_file.write(report)
 
+def build_meta(repo_path: str) -> Dict[str, Any]:
+    """
+    Build the meta JSON object for a file.
+
+    Args:
+        repo_path (str): The path of the repository.
 
-def process_files(paths: Dict[str, Set[Path]], config: Optional[ConfigModel] = None) -> Generator[Tuple[Path, InspectableFile], None, None]:
+    Returns:
+        Dict[str, Any]: The metadata dictionary.
+    """
+    repo_path_obj = Path(repo_path).resolve()
+    git_utils = GIT(repo_path_obj)
+
+    git_data = git_utils.build_git_data()
+    git_metadata = {
```

Comment:
We might want to model the properties in OOD manner i.e.

```
git: {
  "branch": ...
  "commit": ...
  "tag": ...
  "origin": ...
}
```

Reply:
switched